### PR TITLE
Add Jinx

### PR DESCRIPTION
**Pull request by HeyaGlitz**
Also added a combo box to switch between the two.
Diss my code go ahead.
Jinx the cat by https://github.com/SmolFloofyRedhead
in continuation to https://github.com/PolyMC/PolyMC/pull/1495
Signed-off-by: HeyaGlitz <heyaglitz0x99@proton.me>

### CODE REVIEW

**Review by BowDown097 — APPROVED** [review]
W

File: launcher/ui/MainWindow.cpp [hunk 1]
```diff
@@ -1603,26 +1603,36 @@ void MainWindow::setCatBackground(bool enabled)
...
-InstanceView
-{
-    background-image: url(:/backgrounds/%1);
-    background-attachment: fixed;
-    background-clip: padding;
-    background-position: top right;
-    background-repeat: none;
-    background-color:palette(base);
-})").arg(cat));
+        InstanceView
```
Comment (LennyMcLennington) [inline launcher/ui/MainWindow.cpp]: This was unindented for a reason - it's a raw string, any indentation you add will actually be in the string.

Let's focus on functionality changes now and if you want to, you can make a separate PR for formatting changes if you think it's necessary.

File: launcher/ui/pages/global/LauncherPage.ui [hunk 2]
```diff
@@ -38,7 +38,7 @@
       <enum>QTabWidget::Rounded</enum>
      </property>
      <property name="currentIndex">
-      <number>0</number>
+      <number>1</number>
```
Comment (LennyMcLennington) [inline launcher/ui/pages/global/LauncherPage.ui]: Change this to 0 (this happens when you accidentally stay on another tab and then save the UI in designer. Just open designer again and click the first tab and then save the file.)

File: launcher/ui/pages/global/LauncherPage.cpp [hunk 3]
```diff
@@ -430,6 +436,19 @@ void LauncherPage::loadSettings()
...
+        ui->themeComboBoxCat->addItem("BackgroundCat");
+        ui->themeComboBoxCat->addItem("Jinx");
+        if(currentCatStyle == "BackgroundCat")
+        {
+            ui->themeComboBoxCat->setCurrentIndex(0);
+        }
+        else
+        {
+            ui->themeComboBoxCat->setCurrentIndex(1);
+        }
```
Comment (LennyMcLennington) [inline launcher/ui/pages/global/LauncherPage.cpp:450]: I would recommend making it default to BackgroundCat if it reads a non-existent CatStyle from the config.

File: launcher/ui/pages/global/LauncherPage.cpp [hunk 4]
```diff
@@ -363,14 +369,14 @@ void LauncherPage::applySettings()
     // Mods
     s->set("ModMetadataDisabled", ui->metadataDisableBtn->isChecked());
 }
+
```
Comment (LennyMcLennington) [inline launcher/ui/pages/global/LauncherPage.cpp]: Save unrelated formatting changes for another PR

File: launcher/ui/pages/global/LauncherPage.cpp [hunk 5]
```diff
@@ -363,14 +369,14 @@ void LauncherPage::applySettings()
...
 void LauncherPage::loadSettings()
 {
     auto s = APPLICATION->settings();
     // Updates
     if (BuildConfig.UPDATER_ENABLED && APPLICATION->updateChecker()->getExternalUpdater())
     {
-        ui->autoUpdateCheckBox->setChecked(
-                APPLICATION->updateChecker()->getExternalUpdater()->getAutomaticallyChecksForUpdates());
+        ui->autoUpdateCheckBox->setChecked(APPLICATION->updateChecker()->getExternalUpdater()->getAutomaticallyChecksForUpdates());
```
Comment (LennyMcLennington) [inline launcher/ui/pages/global/LauncherPage.cpp]: formatting changes for another PR

File: launcher/ui/pages/global/LauncherPage.cpp [hunk 6]
```diff
@@ -179,6 +179,7 @@ void LauncherPage::on_iconsDirBrowseBtn_clicked()
         ui->iconsDirTextBox->setText(cooked_dir);
     }
 }
+
```
Comment (LennyMcLennington) [inline launcher/ui/pages/global/LauncherPage.cpp]: Save unrelated formatting changes for another PR

File: launcher/resources/backgrounds/backgrounds.qrc [hunk 7]
```diff
@@ -1,8 +1,10 @@
-<!DOCTYPE RCC>
-<RCC version="1.0">
+<RCC>
```
Comment (LennyMcLennington) [inline launcher/resources/backgrounds/backgrounds.qrc]: Why was the header changed?

File: launcher/ui/MainWindow.cpp [hunk 8]
```diff
@@ -1603,26 +1603,36 @@ void MainWindow::setCatBackground(bool enabled)
         QDateTime now = QDateTime::currentDateTime();
         QDateTime birthday(QDate(now.date().year(), 11, 30), QTime(0, 0));
         QDateTime xmas(QDate(now.date().year(), 12, 25), QTime(0, 0));
+
         QString cat;
```
Comment (LennyMcLennington) [inline launcher/ui/MainWindow.cpp]: Recommend guarding against an invalid CatStyle being specified in the config file. Example:
```cpp
QString cat = "default";
QString catStyleSetting = APPLICATION->settings()->get("CatStyle").toString();

if (catStyleSetting == "Jinx")
{
    cat = "jinx";
}
```
(You could write it in a different way, as long as it has the same effect)

**Review by LennyMcLennington — CHANGES_REQUESTED** [review]

File: launcher/resources/backgrounds/backgrounds.qrc (hunk 7 above)
Comment (HeyaGlitz) [reply launcher/resources/backgrounds/backgrounds.qrc]: Idk

**Review by LennyMcLennington — APPROVED** [review]
Everything looks fine to me now

File: launcher/ui/pages/global/LauncherPage.cpp [hunk 9]
```diff
@@ -370,7 +375,7 @@ void LauncherPage::loadSettings()
     if (BuildConfig.UPDATER_ENABLED && APPLICATION->updateChecker()->getExternalUpdater())
     {
         ui->autoUpdateCheckBox->setChecked(
-                APPLICATION->updateChecker()->getExternalUpdater()->getAutomaticallyChecksForUpdates());
```
Comment (LennyMcLennington) [inline launcher/ui/pages/global/LauncherPage.cpp:373]: wait's what's happened here

File: launcher/ui/pages/global/LauncherPage.cpp [hunk 10]
```diff
@@ -370,7 +375,7 @@ void LauncherPage::loadSettings()
...
+                    APPLICATION->updateChecker()->getExternalUpdater()->getAutomaticallyChecksForUpdates());
```
Comment (LennyMcLennington) [inline launcher/ui/pages/global/LauncherPage.cpp]: What's happened here

Comment (HeyaGlitz) [reply launcher/ui/pages/global/LauncherPage.cpp]: just pressed enter, supposed it would go back to how it was. Doubt this really affects the pr

Comment (LennyMcLennington) [reply launcher/ui/pages/global/LauncherPage.cpp]: I don't really want the pr to be modifying some random formatting in a completely unrelated part of the code in the same commit that it's adding specific functionality, it's not good.

**Comment by Capricious-frog** [comment]
Imagine being triggered by a cat. You guys are pathetic.

**Comment by HeyaGlitz** [comment]
> Imagine being triggered by a cat. You guys are pathetic.

if you cared enough to read the code, we are just adding a new cat, not replacing it.

**Comment by LennyMcLennington** [comment]
> Imagine being triggered by a cat. You guys are pathetic.

Imagine being so triggered by PolyMC that you stalk the PRs to insult people without even bothering to read what the PR does xD

**Comment by 62832** [comment]
> Imagine being triggered by a cat.

Are you retarded?

**Comment by SmolFloofyRedhead** [comment]
> Imagine being triggered by a cat. You guys are pathetic.

PFFFFFFHAHAHAHA oh my god dude, it's an option to switch the cat to Jinx. Also Jinx is a meme legend. The neat thing about the way this is being done is that more options for background images could easily be added in the future.

Btw, the current PNGs aren't fully clean. Got fully clean versions on the way, though!

**Comment by HeyaGlitz** [comment]
@LennyMcLennington Everything's ready for merging

**Comment by irisxerno** [comment]
so this is what youve been doing

**Comment by HeyaGlitz** [comment]
> so this is what youve been doing

what?
